pkg/server/plugin/upstreamauthority/spire/spire.go

```diff
@@ -29,10 +39,22 @@ func BuiltIn() catalog.Plugin {
 }
 
 type Plugin struct {
-	mtx sync.RWMutex
-
+	mtx         sync.RWMutex
 	trustDomain url.URL
 	config      *Configuration
+
+	nodeMtx    sync.RWMutex
+	nodeClient node.NodeClient
+	conn       *grpc.ClientConn
+	log        hclog.Logger
+
+	pollMtx                sync.RWMutex
+	stopPolling            context.CancelFunc
+	currentPollSubscribers uint64
+
+	bundleMtx     sync.RWMutex
+	bundleVersion uint
```

Comment:
nit: How would you feel about making this explicitly 64-bit? Even though it will be 64-bit on 64-bit architectures, I'd rather be worried about overflow on a 32-bit platform.
```suggestion
	bundleVersion uint64
```

Reply:
Of course, overflow wouldn't really matter now that I think of it. But it still sets off my spidey sense (32-bit counters, oh my!)

Reply:
sure, will change it 🕷️  👍